code/_onclick/hud/action.dm

```diff
@@ -44,37 +50,28 @@
 	owner = null
 
 /datum/action/proc/Trigger()
-	if(!Checks())
+	if(!IsAvailable())
 		return 0
 	return 1
 
 /datum/action/proc/Process()
 	return
 
-/datum/action/proc/CheckRemoval(mob/living/user) // 1 if action is no longer valid for this mob and should be removed
-	return 0
-
 /datum/action/proc/IsAvailable()
-	return Checks()
-
-/datum/action/proc/Checks()// returns 1 if all checks pass
 	if(!owner)
 		return 0
 	if(check_flags & AB_CHECK_RESTRAINED)
 		if(owner.restrained())
 			return 0
 	if(check_flags & AB_CHECK_STUNNED)
-		if(owner.stunned)
+		if(owner.stunned || owner.weakened)
```

Comment:
if you can't use the button when stunned, you sure as hell can't use it when weakened.